File: Services/api/src/receipts/dto/receipt_dto.rs
use rocket::figment::value::Num;
use rocket::serde::Serialize;

#[derive(Serialize)]
pub struct ReceiptDto {
    pub id: Num,
    pub date: String,
    pub amount_cents: Num,
    pub currency: String,
    pub ext_id: String,
    pub merchant_name: String,
    pub merchant_address: String,
    pub scan_file_name: String,
    pub transaction_id: Option<i32>,
    pub transaction_category: Option<String>,
    pub transaction_amount_cents: Option<i32>,
    pub transaction_currency: Option<String>,
    pub account_code: Option<String>,
    pub account_description: Option<String>,
}

File: Services/api_js/src/modules/receipts/receipt.dto.ts
/**
 * Data Transfer Object for Receipt responses
 */
export interface ReceiptDto {
  id: number;
  date: string;
  amount_cents: number;
  currency: string;
  ext_id: string;
  merchant_name: string;
  merchant_address: string;
  scan_file_name: string;
  transaction_id: number | null;
  transaction_category: string | null;
  transaction_amount_cents: number | null;
  transaction_currency: string | null;
  account_code: string | null;
  account_description: string | null;
}
